hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionRegistries.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.client;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ServiceLoader;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.security.User;
+import org.apache.hadoop.hbase.util.ReflectionUtils;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hbase.thirdparty.com.google.common.collect.ImmutableMap;
+
+/**
+ * The entry point for creating a {@link ConnectionRegistry}.
+ */
+@InterfaceAudience.Private
```

Comment:
Nit: I don't really like this new name.
Could we rename the new ConnectionRegistryFactory interface, and keep the ConnectionFactory name for the implementation ?

Reply:
ConnectionFactory is for creating Connection And AsyncConnection.

Now we have ConnectionRegistryFactory, it is an interface and also used by ServiceLoader to load different ConnectionRegistry implementation. And ConnectionRegistries is used to load the different ConnectionRegistryFactories and create ConnectionRegistry.

So what is your suggested naming pattern? I'm actually neutral on this.

Reply:
Sorry, I meant ConnectionRegistryFactory.

I don't think it's critical either, I'd just prefer to keep the Factory naming.

Maybe the interface/service could be called URLConnectionRegistryFactory or URIConnectionRegistryFactory
and then we could keep the ConnectionRegistryFactory name instead of renaming it to ConnectionRegistries.

Reply:
Let me try.